### PR TITLE
Add `__repr__` for Window class

### PR DESCRIPTION
**Pull request by yunline**
```py
import pygame
from pygame._sdl2 import video

pygame.init()
pygame.display.set_mode((640,480))

win1 = video.Window.from_display_module()
win2 = video.Window("hello window")
win3 = video.Window()
win4 = video.Window()

win4.destroy()

print(win1)
print(win2)
print(win3)
print(win4)
```

```
<Window(From Display)>
<Window(id=2, title='hello window')>
<Window(id=3, title='pygame window')>
<Window(Destroyed)>
```

### CODE REVIEW

**Review by oddbookworm — APPROVED** [review]
LGTM!

File: src_c/window.c [hunk 1]
```diff
@@ -701,6 +701,27 @@ window_from_display_module(PyTypeObject *cls, PyObject *_null)
...
+    title = SDL_GetWindowTitle(self->_win);
+    win_id = SDL_GetWindowID(self->_win);
+    if (win_id == 0) {
+        return RAISE(pgExc_SDLError, SDL_GetError());
+    }
+
+    if (self->_is_borrowed) {
+        return PyUnicode_FromString("<Window(From Display)>");
+    }
+    return PyUnicode_FromFormat("<Window(id=%d, title='%s')>", win_id, title);
```
Comment (Starbuck5) [inline src_c/window.c:722]: Maybe have title first instead of id. Title seems more important.

**Review by Starbuck5 — APPROVED** [review]
Left one note.

Otherwise this looks pretty straightforward. Thanks!

File: src_c/window.c [hunk 2]
```diff
@@ -701,6 +701,27 @@ window_from_display_module(PyTypeObject *cls, PyObject *_null)
...
+
+    title = SDL_GetWindowTitle(self->_win);
+    win_id = SDL_GetWindowID(self->_win);
+    if (win_id == 0) {
+        return RAISE(pgExc_SDLError, SDL_GetError());
+    }
+
+    if (self->_is_borrowed) {
+        return PyUnicode_FromString("<Window(From Display)>");
+    }
```
Comment (ankith26) [inline src_c/window.c:721]: minor nit: should this come above the code that gets `title` and `win_id`, as this codepath doesn't even use those?

Comment (yunline) [reply src_c/window.c]: fixed

**Review by ankith26 — APPROVED** [review]
LGTM now, thanks

**Pull request merged**